src/python/pants/init/engine_initializer.py

```diff
@@ -132,17 +132,14 @@ def _legacy_symbol_table(build_file_aliases: BuildFileAliases) -> SymbolTable:
         tuple(factory.target_types)[0],
       )
 
+  table.update(target_adaptors)
+
   # TODO: The alias replacement here is to avoid elevating "TargetAdaptors" into the public
   # API until after https://github.com/pantsbuild/pants/issues/3560 has been completed.
   # These should likely move onto Target subclasses as the engine gets deeper into beta
   # territory.
-  table['python_library'] = PythonTargetAdaptor
```

Comment:
So I can see these moved to `register.py` and they get loaded in `extension_loader.py`, but I don't really understand the rationale behind this code move. Could you explain? (possibly in the commit message of a distinct commit to help review both changes independently)

Reply:
This code move is the entire rationale for this PR. Right now, the only way to make a BUILD file target name map to a specific `TargetAdaptor` subclass is to put it in `engine_initializer.py`. Outside of the pants codebase (i.e. in the Twitter codebase), we would like to be able to e.g. run Black on our python targets, but right now since *only* the BUILD file string `python_library` is mapped to `PythonTargetAdaptor`, pants in v2 is unable to see any targets in the Twitter monorepo.